### PR TITLE
Purchase Requisition workflow rework

### PR DESCRIPTION
**Pull request by mkoval-ben**

### CODE REVIEW

File: src/zrwtha_resp_cwf_ecc_preq.clas.locals_imp.abap [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    request-user = requester->user_name.
+
+    customer_wf_customizing->get_external_data(
+      EXPORTING
+        iv_object   = 'BEN_DATA'
+        iv_action   = 'KostlFromUser'
+        iv_data     = request
+      IMPORTING
+        ev_data     = response ).
```
Comment (aliakhavets-ben) [inline src/zrwtha_resp_cwf_ecc_preq.clas.locals_imp.abap:95]: are we sure that its not called many times?

File: src/zrwtha_resp_cwf_ecc_preq.clas.locals_imp.abap [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    request-matkl = me->material_group.
+
+    customer_wf_customizing->get_external_data(
+      EXPORTING
+        iv_object   = 'BEN_DATA'
+        iv_action   = 'FreigFromKostlMatkl'
+        iv_data     = request
+      IMPORTING
+        ev_data     = response ).
+    result = response-freig.
```
Comment (aliakhavets-ben) [inline src/zrwtha_resp_cwf_ecc_preq.clas.locals_imp.abap:180]: looks like this one is called even in loop

**Review by aliakhavets-ben — COMMENTED** [review]
@mkoval-ben In general looks good 😃 👍 Especially I liked the smart way how WF is implemented. 
Not sure if I was needed to check everything here. I just went through all.
Please see my all comments linked to the code.

**Comment by mkoval-ben** [comment]
Thanks @aliakhavets-ben! You didn't have to review the entire repo, just the files related to `zrwtha_resp_cwf_ecc_preq`. I realized today that you cannot see the files I marked as viewed for myself, sorry about that.
I will read through your comments and get back to you.

**Comment by mkoval-ben** [comment]
@aliakhavets-ben I've incorporated your suggestions into some parts of the code, mainly removing unused code and changing how `resposibility` is handled. Thank you for your input once again!

**Pull request closed without merging**